lms/services/user.py

```diff
@@ -41,26 +41,23 @@ def upsert_user(self, lti_user: LTIUser) -> User:
         # tied to users in groups. Should we start to store users who have not
         # launched us, we could inflate our numbers or change their meaning.
 
-        user = User(
```

Comment:
Here we were constructing an new User object in all cases, I find that confusing.